### PR TITLE
SONARPY-1139 ApiGateway: RestApi

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java [hunk 1]
```diff
@@ -20,23 +20,43 @@
...
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.HasSymbol;
 
 import static org.sonar.python.checks.cdk.CdkPredicate.isString;
 import static org.sonar.python.checks.cdk.CdkUtils.getArgument;
 
 @Rule(key = "S6333")
 public class PublicApiIsSecuritySensitiveCheck extends AbstractCdkResourceCheck {
-  private static final String SAFE_API_MESSAGE = "Make sure that creating public APIs is safe here.";
+  private static final String ERROR_MESSAGE = "Make sure that creating public APIs is safe here.";
```
Comment (nils-werner-sonarsource) [inline python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java:35]: IMO ERROR is not the right expression for this kind of rule/hotspot. Maybe `ISSUE_MESSAGE` or `MESSAGE` WDYT?

File: python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java [hunk 2]
```diff
@@ -20,23 +20,43 @@
...
     checkFqns(List.of("aws_cdk.aws_apigateway.CfnMethod", "aws_cdk.aws_apigatewayv2.CfnRoute"), (subscriptionContext, callExpression) ->
       getArgument(subscriptionContext, callExpression, "authorization_type").ifPresentOrElse(
-        argument -> argument.addIssueIf(isString("NONE"), SAFE_API_MESSAGE),
+        argument -> argument.addIssueIf(isString("NONE"), ERROR_MESSAGE),
         () -> subscriptionContext.addIssue(callExpression.callee(), OMITTING_MESSAGE)
       )
     );
+    checkFqn("aws_cdk.aws_apigateway.Resource.add_method", (subscriptionContext, callExpression) ->
+      getArgument(subscriptionContext, callExpression, "authorization_type").ifPresent(
+        argument -> argument.addIssueIf(isAuthorizationTypeNone(), ERROR_MESSAGE)
```
Comment (nils-werner-sonarsource) [inline python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java:48]: Isn't `org.sonar.python.checks.cdk.CdkPredicate#isFqn` doing the same job?

File: python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java (hunk 1 above)
Comment (mstachniuk) [reply python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java]: fixed

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**